ckanext/twitter/plugin.py

```diff
@@ -38,17 +34,31 @@ def update_config(self, config):
         # Add resources
         p.toolkit.add_resource('theme/fanstatic', 'ckanext-twitter')
 
-    # IResourceController
+
+    # IResourceController, called after a resource is created
+    def after_create(self, context, res_pkg_dict):
+        data_dict = {'id': res_pkg_dict['package_id']}
+        pkg_dict = tk.get_action('package_show')(context, data_dict)
+
+        is_suitable = twitter_helpers.twitter_pkg_suitable(context,
+                                                           pkg_dict['id'])
+        if is_suitable and ('Twitter_Popup' in pkg_dict.get('twitter_popup', [])):
+            try:
+                session.pop('twitter_is_suitable', '')
+                session.setdefault('twitter_is_suitable', pkg_dict['id'])
+                session.save()
+            except TypeError:
+                print "session not iterable"
+
+
+    # IResourceController, called after a resource is updated
     def after_update(self, context, res_pkg_dict):
-        if ckanext.twitter.lib.config_helpers.tweet_trigger() == 'dataset':
-            pkg_dict = res_pkg_dict
-        else:
-            data_dict = {'id': res_pkg_dict['package_id']}
-            pkg_dict = tk.get_action('package_show')(context, data_dict)
+        data_dict = {'id': res_pkg_dict['package_id']}
```

Comment:
The variable `res_pkg_dict` should probably be `resource` to align with other CKAN code.